curvefs/src/client/rpcclient/metacache.cpp

```diff
@@ -558,18 +555,32 @@ bool MetaCache::GetCopysetInfowithCopySetID(
 
 bool MetaCache::GetPartitionIdByInodeId(uint32_t fsID, uint64_t inodeID,
     PartitionID *pid) {
+    {
+        ReadLockGuard rl(rwlock4Partitions_);
+        for (const auto &it : partitionInfos_) {
+            if (it.start() <= inodeID && it.end() >= inodeID) {
+                *pid = it.partitionid();
+                return true;
+            }
+        }
+    }
+
+    // list form mds
     if (!ListPartitions(fsID)) {
         LOG(ERROR) << "ListPartitions for {fsid:" << fsID
                    << "} fail, partition list not exist";
         return false;
     }
-    ReadLockGuard rl(rwlock4Partitions_);
-    for (const auto &it : partitionInfos_) {
-        if (it.start() <= inodeID && it.end() >= inodeID) {
-            *pid = it.partitionid();
-            return true;
+    {
+        ReadLockGuard rl(rwlock4Partitions_);
+        for (const auto &it : partitionInfos_) {
+            if (it.start() <= inodeID && it.end() >= inodeID) {
+                *pid = it.partitionid();
+                return true;
+            }
```

Comment:
it's repeated, you should extract it to a single function or a lambda

Reply:
fixed